packages/ember-routing/lib/location/history_location.js

```diff
@@ -208,14 +207,14 @@ export default EmberObject.extend({
   onUpdateURL(callback) {
     this._removeEventListener();
 
-    this._popstateHandler = () => {
+    this._popstateHandler = run.bind(this, function() {
       // Ignore initial page load popstate event in Chrome
-      if (!popstateFired) {
-        popstateFired = true;
+      if (!this.__popstateFired) {
+        this.__popstateFired = true;
         if (this.getURL() === this._previousURL) { return; }
       }
       callback(this.getURL());
```

Comment:
I don't think `callback` contains run it starts here https://github.com/emberjs/ember.js/blob/master/packages/ember-routing/lib/system/router.js#L224 and calls various methods in router